frontend/components/Course/Analytics/Cards/SummaryCards.tsx

```diff
@@ -0,0 +1,128 @@
+import { Card, Dropdown, Header, Segment } from "semantic-ui-react";
+import React, { useState } from "react";
+import AnalyticsCard from "./AnalyticsCard";
+import { useStatistic } from "../../../../hooks/data-fetching/analytics";
+import { Metric } from "../../../../types";
+
+interface SummaryCardsProps {
+    courseId: number;
+    queueId: number;
+}
+
+export default function SummaryCards({ courseId, queueId }: SummaryCardsProps) {
+    const [timeRange, setTimeRange] = useState(7);
+    const timeRangeOptions = [7, 30, 90].map((range) => {
+        return {
+            key: range,
+            value: range,
+            text: `Last ${range} days`,
+        };
+    });
+
+    const {
+        data: numAnsweredData,
+        isValidating: numAnsweredValidating,
+    } = useStatistic(courseId, queueId, Metric.NUM_ANSWERED, timeRange);
+
+    const { data: avgWaitData, isValidating: avgWaitValidating } = useStatistic(
+        courseId,
+        queueId,
+        Metric.AVG_WAIT,
+        timeRange
+    );
+
+    const {
+        data: studentsHelpedData,
+        isValidating: studentsHelpedValidating,
+    } = useStatistic(courseId, queueId, Metric.STUDENTS_HELPED, timeRange);
+
+    const {
+        data: avgTimeHelpingData,
+        isValidating: avgTimeHelpingValidating,
+    } = useStatistic(courseId, queueId, Metric.AVG_TIME_HELPING, timeRange);
+
+    const sum = (data) => {
+        return data.reduce((prev, cur) => prev + parseInt(cur.value, 10), 0);
+    };
+
+    const averageWeighted = (data, weights) => {
+        const timeHelping: { value: number; date: string }[] = [];
+        const numHelped: { value: number; date: string }[] = [];
+        for (const dataItem of data) {
+            const weightItem = weights.find(
+                (ele) => dataItem.date === ele.date
+            );
+            if (weightItem) {
+                timeHelping.push({
+                    value: dataItem.value * weightItem.value,
+                    date: dataItem.date,
+                });
+                numHelped.push({
+                    value: weightItem.value,
+                    date: dataItem.date,
+                });
+            }
+        }
+        return sum(timeHelping) / sum(numHelped) || 0;
```

Comment:
Why || 0? If you're checking for divide by 0 errors (which you should definitely do!) I think javascript returns infinity which is not falsy :(

Reply:
interesting, so it looks like js treats 0/0 as NaN which is falsy, but any other numerator as infinity. i will still change it to an explicit check though in the edge case some happens on the backend such that we have a mismatch and timeHelping is positive when numHelped is zero.

Reply:
Sounds good! Also, if you get an error, just log it with sentry and maybe don't display the card - you can return undefined in this helper function and check if average is undefined or not before you display the card. Or display some error state! Not a big deal what you choose, but definitely at least log the error instead of silently returning 0 as the average wait time